contracts/timealarms/src/alarms.rs

```diff
@@ -30,28 +42,52 @@ impl TimeAlarms {
             return Err(ContractError::InvalidAlarm(time));
         }
         contract::validate_addr(&deps.querier, &address)?;
-        Self::TIME_ALARMS.add(deps.storage, address, time)?;
+        self.time_alarms.add(deps.storage, address, time)?;
         Ok(Response::new())
     }
 
     pub fn try_notify(
+        &self,
         storage: &mut dyn Storage,
         ctime: Timestamp,
         max_count: AlarmsCount,
     ) -> Result<Response, ContractError> {
-        let dispatcher = OracleAlarmDispatcher::new();
-        let dispatcher = Self::TIME_ALARMS.notify(storage, dispatcher, ctime, max_count)?;
+        let dispatcher = self
+            .time_alarms
+            .alarms_selection(storage, ctime)
+            .take(max_count.try_into()?)
+            .try_fold(
+                OracleAlarmDispatcher::new(),
+                |dispatcher, alarm| -> Result<OracleAlarmDispatcher, ContractError> {
+                    dispatcher.send_to(Self::REPLY_ID, alarm?.0)
+                },
+            )?;
         dispatcher.try_into()
     }
 
     pub fn try_any_alarm(
+        &self,
         storage: &dyn Storage,
         ctime: Timestamp,
     ) -> Result<AlarmsStatusResponse, ContractError> {
-        Self::TIME_ALARMS
-            .any_alarm(storage, ctime)
-            .map(|remaining_alarms| AlarmsStatusResponse { remaining_alarms })
-            .map_err(Into::into)
+        let remaining_alarms = self
+            .time_alarms
+            .alarms_selection(storage, ctime)
+            .next()
+            .transpose()?
+            .is_some();
+        Ok(AlarmsStatusResponse { remaining_alarms })
+    }
+
+    pub fn migrate_v1(storage: &mut dyn Storage) -> Result<(), ContractError> {
+        AlarmsOld::new(
+            Self::ALARMS_NAMESPACE,
+            Self::ALARMS_IDX_NAMESPACE,
+            "alarms_next_id",
+        )
+        .migrate(storage)?;
+
+        Ok(())
     }
```

Comment:
let's not mix the migration code with the rest

The aim is to locate it in module(s) that are going to be removed once the migration is over.